### PR TITLE
docs: update form docs description

### PR DESCRIPTION
**Pull request by serene-ding**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    docs: update form docs description       |
| 🇨🇳 Chinese |   更新文档描述        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a4c2e66</samp>

Add a note to the `shouldUpdate` prop of the Form.Item component in both English and Chinese documentation, to clarify its usage and avoid confusion. This is based on issue `#34500`.

### <samp>🤖 Generated by Copilot at a4c2e66</samp>

*  Add a note to the `shouldUpdate` prop of the Form.Item component to warn about returning the child component in a function ([link](https://github.com/ant-design/ant-design/pull/44548/files?diff=unified&w=0#diff-28baad7eca32a14feb2ed5fdcee826cb1d27ee818bcbcaa6c9a8ed368c9b16e5L176-R176), [link](https://github.com/ant-design/ant-design/pull/44548/files?diff=unified&w=0#diff-f17ef6685c42eb193b5735088ef3c54e36aca982b8464077915301cabb1d4b8fL165-R168))
*  Link the related issue [#34500](https://github.com/ant-design/ant-design/issues/34500) in the Chinese documentation of the `shouldUpdate` prop ([link](https://github.com/ant-design/ant-design/pull/44548/files?diff=unified&w=0#diff-f17ef6685c42eb193b5735088ef3c54e36aca982b8464077915301cabb1d4b8fL165-R168))

### CODE REVIEW

File: components/form/index.zh-CN.md [hunk 1]
```diff
@@ -162,7 +162,9 @@ const validateMessages = {
 
 Form 通过增量更新方式，只更新被修改的字段相关组件以达到性能优化目的。大部分场景下，你只需要编写代码或者与 [`dependencies`](#dependencies) 属性配合校验即可。而在某些特定场景，例如修改某个字段值后出现新的字段选项、或者纯粹希望表单任意变化都对某一个区域进行渲染。你可以通过 `shouldUpdate` 修改 Form.Item 的更新逻辑。
 
-当 `shouldUpdate` 为 `true` 时，Form 的任意变化都会使该 Form.Item 重新渲染。这对于自定义渲染一些区域十分有帮助：
+当 `shouldUpdate` 为 `true` 时，Form 的任意变化都会使该 Form.Item 重新渲染。这对于自定义渲染一些区域十分有帮助，要注意 Form.Item 里包裹的子组件应以函数形式返回，否则 `shouldUpdate` 行为无效：
```
Comment (yoyo837) [inline components/form/index.zh-CN.md:165]:
```suggestion
当 `shouldUpdate` 为 `true` 时，Form 的任意变化都会使该 Form.Item 重新渲染。这对于自定义渲染一些区域十分有帮助，要注意 Form.Item 里包裹的子组件如果非函数组件，则 `shouldUpdate` 不起作用：
```

这样是不是好一点

File: components/form/index.en-US.md [hunk 2]
```diff
@@ -173,10 +175,10 @@ When `shouldUpdate` is `true`, any Form update will cause the Form.Item to be re
 
 You can ref [example](#components-form-demo-horizontal-login) to see detail.
 
-When `shouldUpdate` is a function, it will be called by form values update. Providing original values and current value to compare. This is very helpful for rendering additional fields based on values:
+When `` is a function, it will be called by form values update. Providing original values and current value to compare. This is very helpful for rendering additional fields based on values.
```
Comment (yoyo837) [inline components/form/index.en-US.md:178]: 这里直接干掉是笔误吗?

File: components/form/index.zh-CN.md (hunk 1 above)
Comment (li-jia-nan) [reply components/form/index.zh-CN.md]: 感觉有歧义了，这里的函数组件知道是 "被函数返回的组件"，但是可能会被用户理解成 "class组件 & function 组件" 中的函数组件？

Comment (yoyo837) [reply components/form/index.zh-CN.md]: suggestion 看看建议

File: components/form/index.zh-CN.md [hunk 3]
```diff
@@ -162,7 +162,9 @@ const validateMessages = {
 
 Form 通过增量更新方式，只更新被修改的字段相关组件以达到性能优化目的。大部分场景下，你只需要编写代码或者与 [`dependencies`](#dependencies) 属性配合校验即可。而在某些特定场景，例如修改某个字段值后出现新的字段选项、或者纯粹希望表单任意变化都对某一个区域进行渲染。你可以通过 `shouldUpdate` 修改 Form.Item 的更新逻辑。
 
-当 `shouldUpdate` 为 `true` 时，Form 的任意变化都会使该 Form.Item 重新渲染。这对于自定义渲染一些区域十分有帮助：
+当 `shouldUpdate` 为 `true` 时，Form 的任意变化都会使该 Form.Item 重新渲染。这对于自定义渲染一些区域十分有帮助，要注意 Form.Item 里包裹的子组件如果非函数组件，则 `shouldUpdate` 不起作用：
```
Comment (li-jia-nan) [inline components/form/index.zh-CN.md:165]:
```suggestion
当 `shouldUpdate` 为 `true` 时，Form 的任意变化都会使该 Form.Item 重新渲染。这对于自定义渲染一些区域十分有帮助，要注意 Form.Item 里包裹的子组件必须由函数返回，否则 `shouldUpdate` 不会起作用：
```

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**